File: Lab_2_1_cpp/BaseString.cpp
#include "BaseString.h"

BaseString::BaseString()
{
    value = "";
}

BaseString::BaseString(std::string val)
{
    value = val;
}

BaseString::BaseString(const BaseString& other)
{
    value = other.value;
}

std::string BaseString::getValue() const
{
    return value;
}

int BaseString::getLength() const
{
    return value.length();
}

File: Lab_2_1_cs/BaseString.cs
using System;

namespace StringLibrary
{
    public class BaseString
    {
        protected string value;

        public BaseString()
        {
            value = "";
        }

        public BaseString(string val)
        {
            value = val;
        }

        public BaseString(BaseString other)
        {
            value = other.value;
        }

        public string GetValue()
        {
            return value;
        }

        public int GetLength()
        {
            return value.Length;
        }
    }
}
